commit 6aa256b6c680245b4c402f2c8a8ee356f03df3ea
Author: Daniel Mack <daniel@zonque.org>
Date:   Sun Oct 19 09:07:36 2014 +0200

    ASoC: fsl: use strncpy() to prevent copying of over-long names
    
    Use strncpy() instead of strcpy(). That's not a security issue, as the
    source buffer is taken from DT nodes, but we should still enforce bound
    checks. Spotted by Coverity.
    
    Signed-off-by: Daniel Mack <daniel@zonque.org>
    Signed-off-by: Mark Brown <broonie@kernel.org>

diff --git a/sound/soc/fsl/fsl_esai.c b/sound/soc/fsl/fsl_esai.c
index 8bcdfda09d7a..a645e296199e 100644
--- a/sound/soc/fsl/fsl_esai.c
+++ b/sound/soc/fsl/fsl_esai.c
@@ -734,7 +734,7 @@ static int fsl_esai_probe(struct platform_device *pdev)
 		return -ENOMEM;
 
 	esai_priv->pdev = pdev;
-	strcpy(esai_priv->name, np->name);
+	strncpy(esai_priv->name, np->name, sizeof(esai_priv->name) - 1);
 
 	/* Get the addresses and IRQ */
 	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);